### PR TITLE
Allow the user to specify a theme

### PR DESCRIPTION
**Pull request by jsalzbergedu**
Some changes necessary to allow the user to specify a theme. Most of it is simple, obvious stuff that would have eventually been put in (like `from_file` constructors). Also added a `WindowBuilder` to allow the user to optionally specify theme etc.

### CODE REVIEW

File: src/theme.rs [hunk 1]
```diff
@@ -30,9 +35,21 @@ impl Theme {
...
+            Ok(_) => Ok(Theme::parse(&css)),
+            Err(err) => Err(err),
+        }
+    }
+
     fn all_rules(&self) -> Vec<Rule> {
         if let Some(ref parent) = self.parent {
-            parent.rules.iter().chain(self.rules.iter()).cloned().collect()
+            self.rules.iter().chain(parent.rules.iter()).cloned().collect()
+            //parent.rules.iter().chain(self.rules.iter()).cloned().collect()
```
Comment (jackpot51) [inline src/theme.rs]: You can remove this comment.

**Comment by jackpot51** [comment]
Looks really good! Just one thing I want removed.

**Comment by jsalzbergedu** [comment]
I changed it. Thank you!

**Comment by jackpot51** [comment]
Sweet! Merged!

**Pull request merged**